File: src/main/scala/im/tox/toktok/app/contacts/FileSendActivity.scala
package im.tox.toktok.app.contacts

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.{ LinearLayoutManager, RecyclerView, Toolbar }
import android.view.MenuItem
import im.tox.toktok.TypedBundleKey._
import im.tox.toktok.TypedResource._
import im.tox.toktok.app.domain.File
import im.tox.toktok.{ BundleKey, R, TR }

import scala.collection.mutable.ListBuffer

final class FileSendActivity extends AppCompatActivity {

  private var mToolbar: Toolbar = null
  private var mRecycler: RecyclerView = null
  private var colorPrimary = 0
  private var colorStatus = 0
  private var userName = ""

  protected override def onCreate(savedInstanceState: Bundle): Unit = {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_files_send)

    val bundle = getIntent.getExtras

    colorPrimary = bundle(BundleKey.contactColorPrimary)
    colorStatus = bundle(BundleKey.contactColorStatus)
    userName = bundle(BundleKey.contactName)

    getWindow.setStatusBarColor(colorStatus)

    mToolbar = this.findView(TR.files_send_toolbar)
    mToolbar.setBackgroundColor(colorPrimary)
    mToolbar.setTitle(getResources.getString(R.string.files_send_title) + " " + userName)

    setSupportActionBar(mToolbar)
    getSupportActionBar.setDisplayHomeAsUpEnabled(true)

    mRecycler = this.findView(TR.files_send_recycler)
    val list = ListBuffer(File.file)

    val mLayoutManager: LinearLayoutManager = new LinearLayoutManager(getBaseContext)
    mRecycler.setLayoutManager(mLayoutManager)

    mRecycler.setAdapter(new FileSendActivityAdapter(list))
  }

  override def onOptionsItemSelected(item: MenuItem): Boolean = {
    item.getItemId match {
      case android.R.id.home =>
        finish()
        true

      case _ =>
        super.onOptionsItemSelected(item)
    }
  }

}

File: sources/app/src/main/java/im/tox/toktok/app/contacts/FileSendActivity.java
package im.tox.toktok.app.contacts;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.Collections;
import java.util.List;

import im.tox.toktok.R;
import im.tox.toktok.app.BundleKey;
import im.tox.toktok.app.domain.File;

public final class FileSendActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_files_send);

        Bundle bundle = getIntent().getExtras();

        int colorPrimary = BundleKey.contactColorPrimary.get(bundle);
        int colorStatus = BundleKey.contactColorStatus.get(bundle);
        String userName = BundleKey.contactName.get(bundle);

        getWindow().setStatusBarColor(colorStatus);

        Toolbar mToolbar = this.findViewById(R.id.files_send_toolbar);
        mToolbar.setBackgroundColor(colorPrimary);
        mToolbar.setTitle(getResources().getString(R.string.files_send_title) + " " + userName);

        setSupportActionBar(mToolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        RecyclerView mRecycler = this.findViewById(R.id.files_send_recycler);
        List<File> list = Collections.singletonList(File.file);

        LinearLayoutManager mLayoutManager = new LinearLayoutManager(getBaseContext());
        mRecycler.setLayoutManager(mLayoutManager);

        mRecycler.setAdapter(new FileSendActivityAdapter(list));
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;

            default:
                return super.onOptionsItemSelected(item);
        }
    }

}
